sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/PrincipalType.java

```diff
@@ -0,0 +1,10 @@
+package org.apache.spark.sql.connector.catalog;
+
+/**
+ * An enumeration support for Role-Based Access Control(RBAC) extensions.
+ */
+public enum PrincipalType {
```

Comment:
this is a public API, please remove it until there is a consensus about adding it.